packages/kotti-ui/source/kotti-modal/KtModal.vue

```diff
@@ -18,37 +18,105 @@
 	</transition>
 </template>
 
-<script>
-export default {
+<script lang="ts">
+import { computed, defineComponent } from '@vue/composition-api'
+
+import { Kotti } from '../types'
+
+export default defineComponent<Kotti.Modal.PropsInternal>({
 	name: 'KtModal',
 	props: {
-		size: {
-			type: String,
-			default: 'md',
-		},
-		closeOutside: {
-			type: Boolean,
-			default: true,
-		},
+		closeOutside: { default: true, type: Boolean },
+		size: { default: Kotti.Modal.Size.MEDIUM, type: String },
 	},
-	computed: {
-		hasBody() {
-			return Boolean(this.$slots['modal-body'])
-		},
-		hasFooter() {
-			return Boolean(this.$slots['modal-footer'])
-		},
-		hasHeader() {
-			return Boolean(this.$slots['modal-header'])
-		},
-		modalClass() {
-			return ['modal-container', `modal-${this.size}`]
-		},
+	setup(props, { emit, slots }) {
+		return {
+			closeModal: () => {
+				if (props.closeOutside) emit('close')
+			},
+			hasBody: computed(() => Boolean(slots['modal-body'])),
+			hasFooter: computed(() => Boolean(slots['modal-footer'])),
+			hasHeader: computed(() => Boolean(slots['modal-header'])),
+			modalClass: computed(() => ['modal-container', `modal-${props.size}`]),
+		}
 	},
-	methods: {
-		closeModal() {
-			if (this.closeOutside) this.$emit('close')
-		},
-	},
-}
+})
 </script>
+
+<style lang="scss">
+@import '../kotti-style/_variables.scss';
+
+.modal-mask {
```

Comment:
can u make the `modal` be the wrapper and use `&-` for practically all of them?

Reply:
Sure